games-util/gamemode/gamemode-1.5.1.ebuild

```diff
@@ -0,0 +1,129 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+MULTILIB_COMPAT=( abi_x86_{32,64} )
+
+inherit meson multilib-minimal ninja-utils systemd
+
+DESCRIPTION="Optimise Linux system performance on demand"
+HOMEPAGE="https://github.com/FeralInteractive/gamemode"
+
+if [[ ${PV} == "9999" ]] ; then
+	EGIT_REPO_URI="https://github.com/FeralInteractive/gamemode.git"
+	GAMEMODE_GIT_PTR="master"
+	inherit git-r3
+else
+	GAMEMODE_GIT_PTR="${PV}"
+	SRC_URI="https://github.com/FeralInteractive/gamemode/releases/download/${GAMEMODE_GIT_PTR}/${P}.tar.xz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="BSD"
+SLOT="0"
+IUSE="systemd"
+
+RDEPEND="
+	acct-group/gamemode
+	sys-apps/dbus[${MULTILIB_USEDEP},systemd=]
+	sys-auth/polkit
+	sys-libs/libcap
+"
+DEPEND="${RDEPEND}"
+
+DOCS="CHANGELOG.md LICENSE.txt README.md example/gamemode.ini"
+
+pkg_pretend() {
+	elog
+	elog "GameMode needs a kernel capable of SCHED_ISO to use its soft realtime"
+	elog "feature. Examples of kernels providing that are sys-kernel/ck-source"
+	elog "and sys-kernel/pf-sources."
+	elog
+	elog "Support for soft realtime is completely optional. It may provide the"
+	elog "following benefits with systems having at least four CPU cores:"
+	elog
+	elog "  * more CPU shares allocated exclusively to the game"
+	elog "  * reduced input lag and reduced thread latency"
+	elog "  * more consistent frame times resulting in less microstutters"
+	elog
+	elog "You probably won't benefit from soft realtime mode and thus don't need"
+	elog "SCHED_ISO if:"
+	elog
+	elog "  * Your CPU has less than four cores because the game may experience"
+	elog "    priority inversion with the graphics driver (thus heuristics"
+	elog "    automatically disable SCHED_ISO usage then)"
+	elog "  * Your game uses busy-loops to interface with the graphics driver"
+	elog "    but you may still force SCHED_ISO per configuation file, YMMV,"
+	elog "    it depends on the graphics driver implementation, i.e. usage of"
+	elog "    __GL_THREADED_OPTIMIZATIONS or similar."
+	elog "  * If your game causes more than 70% CPU usage across all cores,"
+	elog "    SCHED_ISO automatically turns off and on depending on usage and"
+	elog "    is processed with higher-than-normal priority then (renice)."
+	elog "    This auto-switching may result in a lesser game experience."
+	elog
+	elog "For more info look at:"
+	elog "https://github.com/FeralInteractive/gamemode/blob/${GAMEMODE_GIT_PTR}/README.md"
+	elog
+}
+
+multilib_src_configure() {
+	local emesonargs=(
+		-Dwith-systemd=$(usex systemd "true" "false")
```

Comment:
This still doesn't work for me :\
```
 * abi_x86_32.x86: running multilib-minimal_abi_src_configure
meson setup --buildtype plain --libdir lib --localstatedir /var/lib --prefix /usr --sysconfdir /etc --wrap-mode nodownload --cross-file /tmp/portage/games-util/gamemode-1.5.1/temp/meson.i686-pc-linux-gnu.x86 -Dwith-systemd=false -Dwith-systemd-user-unit-dir=/usr/lib/systemd/user -Dwith-examples=false -Dwith-daemon=false /tmp/portage/games-util/gamemode-1.5.1/work/gamemode-1.5.1 /tmp/portage/games-util/gamemode-1.5.1/work/gamemode-1.5.1-abi_x86_32.x86
...
...
Run-time dependency libsystemd found: NO (tried pkgconfig and cmake)
meson.build:83:0: ERROR: Dependency "libsystemd" not found, tried pkgconfig and cmake
```

Weird that they have a systemd flag if it's not optional after all...?

Reply:
Hmm, yes... I'll report upstream. Does it work correctly for you if you don't specify `--with-systemd` at all? Maybe I should setup a container without systemd, tho that's not strictly supported with systemd containers (but it kinda works).

Reply:
Nope, I don't really have the time to dig into their meson.build but I'm sure there's some sort of small mistake that could be patched. However if it looks like a huge task, then it's better to make this depend on systemd for now I guess.

Reply:
Okay, I'll let it depend on systemd for now and work with upstream on fixing it. Next revbump should then support non-systemd properly.

Reply:
I've files a bug report upstream and temporarily worked around it by setting `REQUIRED_USE`, hope that's ok.